docs/csharp/tutorials/nullable-reference-types.md

```diff
@@ -0,0 +1,225 @@
+
+```csharp
+#nullable enable
+namespace NullableIntroduction
+{
+    public class SurveyResponse
+    {
+        public int Id { get; }
+
+        public SurveyResponse(int id) => Id = id;
+    }
+}
+```
+
+Next, add a `static` method to create new participants by generating a random id:
+
+[!code-csharp[GenerateRespondents](../../../samples/csharp/NullableIntroduction/NullableIntroduction/SurveyResponse.cs#Random)]
+
+The main responsibility of this class is to generate the responses for a participant to the questions in the survey. This responsibility has a few steps:
+
+1. Ask for participation in the survey. If the person doesn't consent, return a missing (or null) response.
+1. Ask each question and record the answer. Each answer may also be missing (or null).
+
+Add the following code to your `SurveyRespondent` class:
+
+[!code-csharp[AnswerSurvey](../../../samples/csharp/NullableIntroduction/NullableIntroduction/SurveyResponse.cs#AnswerSurvey)]
+
+The storage for the survey answers is a `Dictionary<int, string>?`, indicating that it may be null. You're using the new language feature to declare your design intent, both to the compiler and to anyone reading your code later. If you ever dereference `surveyResponses` without checking for the null value first, you'll get a compiler warning. You don't get a warning in the `AnswerSurvey` method because the compiler can determine the `surveyResponses` variable was set to a non-null value above.
+
+Next, you need to write the `PerformSurvey` method in the `SurveyRun` class. Add the following code in the `SurveyRun` class:
+
+[!code-csharp[PerformSurvey](../../../samples/csharp/NullableIntroduction/NullableIntroduction/SurveyRun.cs#PerformSurvey)]
+
+Here again, your choice of a nullable `List<SurveyResponse>?` indicates the response may be null. That indicates the survey hasn't been given to any respondents yet. Notice that respondents are added until enough have consented.
+
+The last step to run the survey is to add a call to perform the survey at the end of the `Main` method:
+
+[!code-csharp[RunSurvey](../../../samples/csharp/NullableIntroduction/NullableIntroduction/Program.cs#RunSurvey)]
+
+## Examine survey responses
+
+The last step is to display survey results. You'll add code to many of the classes you've written. This code demonstrates the value of distinguishing nullable and non-nullable reference types. Start by adding the following two expression bodied members to the `SurveyResponse` class:
+
+[!code-csharp[ReportResponses](../../../samples/csharp/NullableIntroduction/NullableIntroduction/SurveyResponse.cs#SurveyStatus)]
+
+Because `surveyResponses` is a non-nullable reference type no checks are necessary before de-referencing it. The `Answer` method returns a non-nullable string, so choose the overload of `GetValueOrDefault` that takes a second argument for the default value.
```

Comment:
```suggestion
Because `surveyResponses` is a non-nullable reference type no checks are necessary before de-referencing it. The `Answer` method returns a non-nullable string, so choose the overload of `GetValueOrDefault` that takes the second argument for the default value.
```

Reply:
I'll leave this as is.